org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/services/XMLRename.java

```diff
@@ -169,11 +172,58 @@ else if(node.isElement()) {
 					}
 				}
 			}
+
+			if (XSDUtils.isXSComplexType(element)) {
+
+				try {
+					List<DOMAttr> attributes = element.getAttributeNodes();
+					DOMAttr attrForValueRename = getAttributeForValueRename(xmlDocument, attributes, position);
+					
+					if (attrForValueRename.getName().equals("name")) {
+						List<Location> locations = getReferenceLocations(node);
+						return renameAttributeValueTextEdits(xmlDocument, attrForValueRename, newText, locations);
+					}
+				
+				} catch (NullPointerException e ) {
```

Comment:
Why this catch? If you have a NPE, you should fix it before.

Reply:
I thought it might be cleaner in this case, because `getAttributeNodes()` on line 179 might return null and `getAttributeForValueRename()` on line 180 might return null. 

Should I just do `if (attributes == null)` and `if (attrForValueRename == null)` checks instead?

Reply:
getAttributeNodes should never return null. It should be fixed by returning a static empty array. See EMPTY_CHILDREN in the DOMNode, I had the same problem.